src/domain/synthetics/trade/usePositionEditorState.ts

```diff
@@ -1,44 +1,49 @@
 
-import { getSyntheticsCollateralEditAddressKey, getSyntheticsCollateralEditAddressMapKey } from "config/localStorage";
+import type { ContractsChainId, SourceChainId } from "config/chains";
+import {
+  getSyntheticsCollateralEditAddressMapKey,
+  getSyntheticsCollateralEditTokenIsFromGmxAccountMapKey,
+} from "config/localStorage";
+import { isSettlementChain } from "config/multichain";
+import { useSettings } from "context/SettingsContext/SettingsContextProvider";
 import { useLocalStorageSerializeKey } from "lib/localStorage";
-import { SUPPORTED_CHAIN_IDS } from "sdk/configs/chains";
-import { getTokens } from "sdk/configs/tokens";
 
 import { parsePositionKey } from "../positions";
-
 export type PositionEditorState = ReturnType<typeof usePositionEditorState>;
 
-// start script to migrate getSyntheticsCollateralEditAddressKey to getSyntheticsCollateralEditAddressMapKey
-
-for (const chainId of SUPPORTED_CHAIN_IDS) {
-  const mapKey = getSyntheticsCollateralEditAddressMapKey(chainId);
-  const alreadyExists = localStorage.getItem(JSON.stringify(mapKey));
-  if (alreadyExists) {
-    continue;
-  }
-  const map = {};
-  for (const token of getTokens(chainId)) {
-    const key = getSyntheticsCollateralEditAddressKey(chainId, token.address);
-    const rawValue = localStorage.getItem(JSON.stringify(key));
-    if (!rawValue) {
-      continue;
-    }
-    map[token.address] = JSON.parse(rawValue);
-
-    localStorage.removeItem(JSON.stringify(key));
-  }
-
-  localStorage.setItem(JSON.stringify(mapKey), JSON.stringify(map));
-}
-
-export function usePositionEditorState(chainId: number) {
+export function usePositionEditorState(chainId: ContractsChainId, srcChainId: SourceChainId | undefined) {
+  // const expressOrdersEnabled = useSelector(selectExpressOrdersEnabled);
+  const { expressOrdersEnabled } = useSettings();
   const [editingPositionKey, setEditingPositionKey] = useState<string>();
   const [collateralInputValue, setCollateralInputValue] = useState("");
   const [selectedCollateralAddressMap, setSelectedCollateralAddressMap] = useLocalStorageSerializeKey<
     Partial<Record<Address, Address>>
   >(getSyntheticsCollateralEditAddressMapKey(chainId), {});
+  const [_isCollateralTokenFromGmxAccount, _setIsCollateralTokenFromGmxAccount] = useLocalStorageSerializeKey<boolean>(
+    getSyntheticsCollateralEditTokenIsFromGmxAccountMapKey(chainId),
+    false
+  );
+
+  let isCollateralTokenFromGmxAccount = false;
+  if (srcChainId !== undefined) {
+    isCollateralTokenFromGmxAccount = true;
+  } else if (!isSettlementChain(chainId)) {
+    isCollateralTokenFromGmxAccount = false;
+  } else {
+    isCollateralTokenFromGmxAccount = Boolean(_isCollateralTokenFromGmxAccount);
+  }
+
+  let setIsCollateralTokenFromGmxAccount: (value: boolean) => void;
+  if (srcChainId !== undefined) {
+    setIsCollateralTokenFromGmxAccount = noop;
+  } else if (!isSettlementChain(chainId)) {
+    setIsCollateralTokenFromGmxAccount = noop;
+  } else {
```

Comment:
looks like only one condition is enough to determine if setIsCollateralTokenFromGmxAccount should be settled

Also in the condition above